pkg/prometheus/promcfg.go

```diff
@@ -2320,6 +2329,9 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 
 	if sc.Spec.RelabelConfigs != nil {
 		relabelings = append(relabelings, generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.RelabelConfigs))...)
+	}
+
+	if sc.Spec.JobName != "" || sc.Spec.RelabelConfigs != nil {
```

Comment:
```suggestion
	if len(relabelings) > 0 {
```

Reply:
I created two branches:
* feature/crd-scrapeConfig-with-jobName-relabelling-0
* feature/crd-scrapeConfig-with-jobName-relabelling-1

The digit in the end is the same as the comparison used in line 2334 for.

The issue: the suggested refactoring is not equal. 
The variable relabelings is already of `length=1` when initialized.
→ Setting the relabeling to `>0` leaves to a ton of changes of other files not relevant to this feature.
→ Setting the relabeling to `>1` requires only one change to the feature-specific test-file with empty relabel config. 

I'd think the second option is better continuing the current behavior.
Yet, I am familiar enough with the overall behavior to evaluate if the overall change with option one is more desirable...

Just let me know which way you prefer and I merge the one into this merge request branch.
And Merry Christmas ;)

Reply:
I see your concern but I think that the current implementation is prone to error. I've submitted #6201 to "fix" it.

Reply:
Thanks for your fix. With your refactoring the question of >0 or >1 is obsolete...